File: 4.8-bc/py/stage-3-hexagonal/app/shared/httputil.py
"""Shared HTTP helpers — error mapping AppError -> HTTPException.

Cross-cutting utilities only. No business logic, no BC-specific imports.
"""
from __future__ import annotations

from fastapi import HTTPException

from app.shared.apperr import AppError


def to_http_exception(err: Exception) -> HTTPException:
    if isinstance(err, AppError):
        return HTTPException(
            status_code=err.status_code,
            detail={"error": err.code, "message": err.message},
        )
    return HTTPException(
        status_code=500,
        detail={"error": "internal", "message": "unexpected error"},
    )

File: 4.8-bc/ts/stage-3-hexagonal/src/shared/httputil.ts
// shared/httputil — спільні HTTP helpers (JSON write + error mapping).
// Cross-cutting утиліти, мінімум — без бізнес-логіки.
import type { FastifyReply } from "fastify";

import { AppError, isAppError } from "./apperr.js";

export function writeError(reply: FastifyReply, err: unknown): FastifyReply {
  if (isAppError(err)) {
    return reply
      .code(err.statusCode)
      .send({ error: err.code, message: err.message });
  }
  return reply
    .code(500)
    .send({ error: "internal", message: "unexpected error" });
}

export { AppError };
